aviatrix/resource_aviatrix_transit_gateway_peering.go

```diff
@@ -89,6 +89,16 @@ func resourceAviatrixTransitGatewayPeering() *schema.Resource {
 				ForceNew:    true,
 				Description: "(Optional) Enable peering over private network. Insane mode is required on both transit gateways. Available as of provider version R2.17.1",
 			},
+			"enable_single_tunnel_mode": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				ForceNew: true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					return !d.Get("enable_peering_over_private_network").(bool)
+				},
```

Comment:
Why is this DiffSuppressFunc needed?

Reply:
If it's not peering over private network, there's only one tunnel anyway. So enabling the single_tunnel or not doesn't matter.

Reply:
👍 Makes sense so if `enable_peering_over_private_network` is false then we don't care about the value of `enable_single_tunnel_mode`. I like this way of doing it, seems very clean!